node/blockchain/views/signed_change_request.py

```diff
@@ -1,15 +1,45 @@
+from rest_framework import status
 from rest_framework.mixins import CreateModelMixin
+from rest_framework.response import Response
 from rest_framework.viewsets import GenericViewSet
 
+from node.blockchain.facade import BlockchainFacade
+from node.blockchain.models import Block
 from node.blockchain.serializers.signed_change_request import SignedChangeRequestSerializer
+from node.core.exceptions import DRFValidationErrorConverter, ValidationError
+from node.core.utils.cryptography import get_signing_key
+from node.core.utils.types import Type
 
 
 class SignedChangeRequestViewSet(CreateModelMixin, GenericViewSet):
     serializer_class = SignedChangeRequestSerializer
 
     def perform_create(self, serializer):
-        instance = serializer.save()
+        signed_change_request = serializer.save()
+        blockchain_facade = BlockchainFacade.get_instance()
 
-        # TODO(dmu) CRITICAL: Provide actual implementation of perform_create()
-        #                     https://thenewboston.atlassian.net/browse/BC-167
-        raise NotImplementedError(instance.json())
+        if Type(serializer.validated_data['message']['type']) == Type.NODE_DECLARATION:
```

Comment:
why do we limit to node declaration only?

but I think we need to restrict sending genesis signed change requests though

Reply:
Updated. Also API return status_code 400 when Type is GENESIS.